oqsprov/oqsprov_keys.c

```diff
@@ -33,26 +35,110 @@ void oqsx_freeprovctx(PROV_OQS_CTX *ctx) {
 
 /// Key code
 
-OQSX_KEY *oqsx_key_new(OSSL_LIB_CTX *libctx, char* oqs_name, char* tls_name, int is_kem, const char *propq)
+static const OQSX_KEX_INFO nids_ecp[] = {
+        { EVP_PKEY_EC, NID_X9_62_prime256v1, 0, 65 , 121, 32}, // level 1
```

Comment:
Is there really no other way to reference to classic crypto than by NID? I thought one of the goals of OSSL3 was to get rid of them (?). But again, this is a nitpick and surely asking too much in this context.

Reply:
I was careful to not use any internal or deprecated OSSL API, but the API used should still be backward compatible with OSSL111. Will still have to dig to the new provider API to see if there is another way, e.g. with strings or OIDs.
Defining the key-, secret-, and ciphertext-sizes is also not optimal yet. I'd prefer to be able to query them via some provider API than defining them here.